### PR TITLE
FIX : missing hook initialization (IMPORTANT)

### PR DESCRIPTION
**Pull request by altairis-tof**
getSellPrice hook was not triggered because of missing initialization

### CODE REVIEW

File: htdocs/product/class/product.class.php [hunk 1]
```diff
@@ -1814,6 +1814,7 @@ public function getSellPrice($thirdparty_seller, $thirdparty_buyer, $pqp = 0)
 
 		// Call hook if any
 		if (is_object($hookmanager)) {
+			$hookmanager->initHooks(array('productdao'));
```
Comment (eldy) [inline htdocs/product/class/product.class.php:1817]: A hook must not be intialized by classes. Hooks must be initialized in main cases by the main parent page or the main CLI script.
Module using the hook must use the context of the page.

Comment (eldy) [inline htdocs/product/class/product.class.php:1817]: We have some initHook(array('objectdao')) inside some method of class.
But this is a bad practice. A context of use should not be defined into a class. It must be defined by the parent page.
If a module need to have its hook executed in all cases, he can use the context 'all' or 'main' (all web pages).

Comment (altairis-tof) [reply htdocs/product/class/product.class.php]: ok @eldy i have update my PR and i initialize the hook in takepos/invoice.php (https://github.com/Dolibarr/dolibarr/pull/27801/commits/f69ed514cd95d7b766cb4b1bc5aa4620d50123e5)
(there are other hook init in product.class.php it is why i did like that)

**Comment by altairis-tof** [comment]
forgot it, i will test the takeposinvoice context in my module !
sorry for the inconvenience

**Pull request closed without merging**